scalding-core/src/main/scala/com/twitter/scalding/serialization/JavaStreamEnrichments.scala

```diff
@@ -213,17 +228,15 @@ object JavaStreamEnrichments {
      */
     def writePosVarInt(i: Int): Unit = {
       require(i >= 0, s"must be non-negative: ${i}")
-      if (i < ((1 << 8) - 1)) s.write(i.toByte)
+      if (i < ((1 << 8) - 1)) s.write(i)
```

Comment:
require is defined as:
`require(requirement: Boolean, message: ⇒ Any)`

Does that mean that the string interpolation won't happen unless `i<0`? Does it also mean that every time this method is called, a closure is created to hold the string interpolation?

We found a performance regression in parquet due to something similar, but using a not-lazy method call (plain java method) and found that we were doing expensive string interpolation for every call, not just in the error condition.